controllers/azurecluster_reconciler.go

```diff
@@ -162,18 +162,19 @@ func (s *azureClusterService) Delete(ctx context.Context) error {
 	if !ShouldDeleteIndividualResources(ctx, s.scope) {
 		// If the resource group is managed, delete it.
 		// We need to explicitly delete vnet peerings, as it is not part of the resource group.
-		vnetPeeringsSvc, err := s.getService(vnetpeerings.ServiceName)
+		vnetPeeringsSvc, err := getService(vnetpeerings.ServiceName, s.services)
```

Comment:
If we already have our set of services locally, why are we converting this from a private method into a func?

Reply:
Good catch, at some point I refactored it to be able to use it across different reconcilers but that ended up not being necessary, will revert

Reply:
Done!